### DISCUSSION TITLE
My editor loses focus while typing

### DISCUSSION THREAD

**Discussion by GuyoST** [react-page/react-page#1150] [Q&A] [answered]
Hello every one, 

I just installed react-page and followed the quick start with slate() and image plugin only, so my component is very basic : 

`return (
        <>
            <Editor cellPlugins={cellPlugins as any} value={value} onChange={setValue} />
        </>
    )`

and of course I also have : 
`const cellPlugins = [slate(), image];
const [value, setValue] = useState<Value>(null!);   //Valeur pour l'éditeur ReactPage`

I imported the css correctly.

Everything is showing well, I can add the "+" icon and select add a Text block for example.

If I start typing some text, after few seconds or randomly, it loses the focus and I can't type anymore. Sometimes it also rollback and restore the text that was just a moment before.

I notice that if I remove `onChange={setValue}` the problem doesn't happen, but of course I don't have the value saved.

Does someone has the same trouble ? or have a solution for that ?

Thank you very much

**Comment by macrozone** [comment, marked as answer]
@GuyoST a reproduction would be nice.

notice that `cellPlugins = [slate(), image];` will create new objects on each rerender, wich might lead to the problem. Can you try memooize that?  or delcare it as constant outside of your component?

```
const cellPlugins = useMemo(() =>  [slate(), image], []);

```

side note: cellPlugins should not require an `any` cast, but i noticed that it might complain when you use ts strict mode. we still have an open pull request to fix that, i should finish that soon...

**Reply by GuyoST** [reply]
Hello !!
Thank you so much for this fast reply.
It's working with `useMemo` !

And yes for the cast, I have to cast to `any` because it complains with typescript.
I will remove the cast as soon as it's not needed any more.
![bitmoji](https://sdk.bitmoji.com/render/panel/2eb3b5b8-3470-474a-b5f1-70438c9ad396-4737f92b-9c14-4349-87c1-1a21b1b0e2e3-v1.png?transparent=1&palette=1&width=246)

**Reply by macrozone** [reply]
nice!

for the typescript issue, follow this issue: https://github.com/react-page/react-page/issues/1104
